### DISCUSSION TITLE
Prisma -> Supabase timeout

### DISCUSSION THREAD

**Discussion by joaolandino** [prisma/prisma#19328] [Q&A] [answered]
Hey guys!

I deployed a site at Vercel, using Next.js and Prisma as ORM. My Postgres database is on Supabase.

I'm having timeout problems in my application, even with the session request (next-auth).

Could you tell me what could be going on? I know that the timeout is 10 seconds, but I still don't want to increase that timeout because 10 seconds is already too much.

https://github.com/prisma/prisma/assets/26358607/e79882f3-e066-4a15-b95e-615771b44b9d

Thanks!

**Comment by janpio** [comment, marked as answer]
What is your video showing exactly? I can't really make sense of it.

If anything takes >10s with your database, it is clearly to far away from your application. Make sure your serverless function on Vercel runs close to your Supabase database.

**Reply by joaolandino** [reply]
Hi @janpio! I made two tweaks that helped a lot! First my Vercel and Supabase were in the Brazil region. I changed both to east-1. Another adjustment I made was changing the node version, which was in version 18 and I changed it to 14.

This solved the 10 second problem.

Thanks!
